File: app/src/main/java/app/xunxun/homeclock/MyApplication.java
package app.xunxun.homeclock;

import android.app.Activity;
import android.app.Application;

import com.crashlytics.android.Crashlytics;
import com.umeng.analytics.MobclickAgent;

import io.fabric.sdk.android.Fabric;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by fengdianxun on 2017/1/24.
 */

public class MyApplication extends Application{
    private List<Activity> activities = new ArrayList<>();
    @Override
    public void onCreate() {
        super.onCreate();
        MobclickAgent.setCatchUncaughtExceptions(false);
        Fabric.with(this, new Crashlytics());
    }

    public void pushActivity(Activity activity){
        activities.add(activity);
    }

    public void popActivity(Activity activity){
        activities.remove(activity);
    }

    public void clearActivities(){
        for (Activity activity: activities) {
            activity.finish();
        }
        activities.clear();
    }
}

File: app/src/main/java/app/xunxun/homeclock/MyApplication.kt
package app.xunxun.homeclock

import android.app.Activity
import android.app.Application

//import com.crashlytics.android.Crashlytics;
import com.crashlytics.android.Crashlytics
import com.umeng.analytics.MobclickAgent

//import io.fabric.sdk.android.Fabric;
import java.util.ArrayList

import io.fabric.sdk.android.Fabric

/**
 * Created by fengdianxun on 2017/1/24.
 */

class MyApplication : Application() {
    private val activities = ArrayList<Activity>()
    override fun onCreate() {
        super.onCreate()
        MobclickAgent.setCatchUncaughtExceptions(false)
        Fabric.with(this, Crashlytics())
    }

    fun pushActivity(activity: Activity) {
        activities.add(activity)
    }

    fun popActivity(activity: Activity) {
        activities.remove(activity)
    }

    fun clearActivities() {
        for (activity in activities) {
            activity.finish()
        }
        activities.clear()
    }
}
